app/models/poll.rb

```diff
@@ -55,9 +62,25 @@ def options_in_rank
     current_state.each.with_index.each_with_object(Hash.new { |h, k| h[k] = [] }) { |(x, i), hash| hash[x] << options.ids[i] }
   end
 
+  def max_voters
+    self[:max_voters] || Float::INFINITY
+  end
+
   private
 
+  def calculate_ranks
+    SchulzeBasic.do(vote_results, vote_results.first.count).ranks
+  end
+
   def ensure_status_is_correct
     draft! if options.empty?
   end
+
+  def max_voters_should_be_number
+    errors.add(:max_voters, 'should be number greater than 0') unless (is_a?(Integer) || Float::INFINITY) && max_voters > 0
```

Comment:
It never equal infinity, actually. According to the normal flow user never can send infinity to this field, so we care only for that case when, for some reason, database in production changed from postgress and user manually construct request with infinity value and get error. Not a big deal, isn't? More than that, I thought that object can have Infinity in that field before saving, but somehow is not, look at this:
![poll_max_voters](https://cloud.githubusercontent.com/assets/13506257/19191191/55a837da-8cb2-11e6-85e0-c213914a1818.png)